tests/rptest/services/trino_service.py

```diff
@@ -0,0 +1,105 @@
+# by the Apache License, Version 2.0
+
+import os
+
+from ducktape.services.service import Service
+from ducktape.cluster.cluster import ClusterNode
+from ducktape.utils.util import wait_until
+import subprocess
+
+
+class TrinoService(Service):
+    """Trino service for querying data generated in datalake."""
+
+    ICEBERG_CONNECTOR_CONF = """
+connector.name=iceberg
+iceberg.catalog.type=rest
+iceberg.rest-catalog.uri={catalog_rest_uri}
+fs.native-s3.enabled=true
+s3.endpoint={s3_endpoint}
+s3.region={s3_region}
+s3.path-style-access=true
+s3.aws-access-key={s3_access_key}
+s3.aws-secret-key={s3_secret_key}"""
+
+    TRINO_HOME = "/opt/trino"
+    TRINO_CONF_PATH = os.path.join(
+        TRINO_HOME, "/opt/trino/etc/catalog/redpanda.properties")
+    PERSISTENT_ROOT = "/var/lib/trino"
+    LOG_FILE = os.path.join(PERSISTENT_ROOT, "trino_server.log")
+    logs = {"iceberg_rest_logs": {"path": LOG_FILE, "collect_default": True}}
+
+    def __init__(self,
+                 ctx,
+                 iceberg_catalog_rest_uri: str,
+                 cloud_storage_access_key: str = 'panda-user',
+                 cloud_storage_secret_key: str = 'panda-secret',
+                 cloud_storage_region: str = 'panda-region',
+                 cloud_storage_api_endpoint: str = "http://minio-s3:9000",
+                 node: ClusterNode | None = None):
+        super(TrinoService, self).__init__(ctx, num_nodes=0 if node else 1)
+        self.iceberg_catalog_rest_uri = iceberg_catalog_rest_uri
+        self.cloud_storage_access_key = cloud_storage_access_key
+        self.cloud_storage_secret_key = cloud_storage_secret_key
+        self.cloud_storage_region = cloud_storage_region
+        self.cloud_storage_api_endpoint = cloud_storage_api_endpoint
+        self.trino_url = None
+
+    def start_node(self, node, timeout_sec=60, **kwargs):
+        node.account.ssh(f"mkdir -p {TrinoService.PERSISTENT_ROOT}")
+        node.account.ssh(f"rm -f {TrinoService.TRINO_CONF_PATH}")
+        connector_config = TrinoService.ICEBERG_CONNECTOR_CONF.format(
+            catalog_rest_uri=self.iceberg_catalog_rest_uri,
+            s3_endpoint=self.cloud_storage_api_endpoint,
+            s3_region=self.cloud_storage_region,
+            s3_access_key=self.cloud_storage_access_key,
+            s3_secret_key=self.cloud_storage_secret_key)
+        self.logger.debug(f"Using connector config: {connector_config}")
+        node.account.create_file(TrinoService.TRINO_CONF_PATH,
+                                 connector_config)
+        node.account.ssh(
+            f"nohup /opt/trino/bin/trino-launcher run 1> {TrinoService.LOG_FILE} 2>&1 &",
+            allow_fail=False)
+        self.trino_url = f"http://{node.account.hostname}:8083"
+        self.wait(timeout_sec=30)
+
+    def wait_node(self, node, timeout_sec=None):
+        check_cmd = f"java -jar /opt/trino/trino-rest-client.jar {self.trino_url} --execute='show catalogs'"
```

Comment:
Consider using: https://github.com/trinodb/trino-python-client instead of executing Java and parsing output. From the experience i can tell the output parsing is usually much less reliable